### PR TITLE
feat(react-avatar): add `shape` into avatar context

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
Add `shape` into AvatarContext.
This is useful when other components needs square shaped avatar.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                            |     Baseline (minified/GZIP) |                           PR |                                                                                                                                                                                                 Change |
| :----------------------------------------------------------------------------------------------------------- | ---------------------------: | ---------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                      | `93.779 kB`<br />`22.493 kB` | `93.799 kB`<br />`22.502 kB` |  `20 B` <br />`9 B`  |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                   | `57.544 kB`<br />`14.966 kB` | `57.564 kB`<br />`14.976 kB` | `20 B` <br />`10 B`  |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr> | `73.758 kB`<br />`19.478 kB` | `73.778 kB`<br />`19.491 kB` | `20 B` <br />`13 B`  |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                | `64.465 kB`<br />`16.892 kB` |   `64.485 kB`<br />`16.9 kB` |  `20 B` <br />`8 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |   `15.632 kB`<br />`6.258 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |   `65.31 kB`<br />`17.931 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> |  `204.34 kB`<br />`57.141 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |    `36.086 kB`<br />`11.9 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.446 kB`<br />`2.186 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `147.659 kB`<br />`40.567 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `44.348 kB`<br />`12.347 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `130.218 kB`<br />`33.128 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |  `78.132 kB`<br />`19.127 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `77.462 kB`<br />`18.939 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/ee7173e0feabf921a7a8a7cbbb0e63afa4f83f84'>ee7173e0feabf921a7a8a7cbbb0e63afa4f83f84</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)

  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/InfoButton-mount_base.html">20</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/InfoButton-mount.html">19</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr></table><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Avatar-mount_base.html">627</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Avatar-mount.html">626</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Button-mount_base.html">372</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Button-mount.html">380</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Field-mount_base.html">1136</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Field-mount.html">1258</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProvider-mount_base.html">839</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProvider-mount.html">871</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProviderWithTheme-mount_base.html">113</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProviderWithTheme-mount.html">114</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">99</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">106</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">111</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">110</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/InfoButton-mount_base.html">20</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/InfoButton-mount.html">19</a></td>
            <td>5000</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/InfoButton-mount.regression.txt">Possible regression</a></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/MakeStyles-mount_base.html">989</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/MakeStyles-mount.html">1016</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Persona-mount_base.html">2031</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/Persona-mount.html">1768</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/SpinButton-mount_base.html">1556</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/27738/perf-test-react-components/SpinButton-mount.html">1513</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

**Review by sopranopillow — APPROVED** [review]
Looks good!

File: packages/react-components/react-avatar/src/contexts/AvatarContext.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { AvatarSize } from '../components/Avatar/Avatar.types';
+import { AvatarSize, AvatarShape } from '../components/Avatar/Avatar.types';
```
Comment (khmakoto) [inline packages/react-components/react-avatar/src/contexts/AvatarContext.ts:2]: _nit:_

```suggestion
import type { AvatarShape, AvatarSize } from '../components/Avatar/Avatar.types';
```

File: packages/react-components/react-avatar/src/index.ts [hunk 2]
```diff
@@ -5,8 +5,16 @@ export {
...
-export type { AvatarNamedColor, AvatarProps, AvatarSlots, AvatarState, AvatarSizes, AvatarSize } from './Avatar';
+export type {
+  AvatarNamedColor,
+  AvatarProps,
+  AvatarSlots,
+  AvatarState,
+  // eslint-disable-next-line deprecation/deprecation
+  AvatarSizes,
+  AvatarSize,
+  AvatarShape,
```
Comment (khmakoto) [inline packages/react-components/react-avatar/src/index.ts:16]: _nit:_ Keep alphabetical order

```suggestion
  AvatarShape,
  // eslint-disable-next-line deprecation/deprecation
  AvatarSizes,
  AvatarSize,
```

**Pull request merged**